test/spec/challengesCtrlSpec.js

```diff
@@ -0,0 +1,174 @@
+'use strict';
+
+describe('Challenges Controller', function() {
+  var $rootScope, scope, user, ctrl, challenges, groups;
+
+  beforeEach(function() {
+    module(function($provide) {
+      $provide.value('User', {});
+    });
+
+    inject(function($rootScope, $controller, Challenges, Groups){
+      user = specHelper.newUser();
+      user._id = "unique-user-id";
+
+      scope = $rootScope.$new();
+
+      // Load RootCtrl to ensure shared behaviors are loaded
+      $controller('RootCtrl',  {$scope: scope, User: {user: user}});
+
+      ctrl = $controller('ChallengesCtrl', {$scope: scope, User: {user: user}});
+
+      challenges = Challenges;
+
+      groups = Groups;
+    });
+  });
+
+  it('filters challenges', function() {
+
+   //There are for types of challenges
+
+   //You can be the owner and a member
+   var ownMem = new challenges.Challenge({
+     name: 'test',
+     description: 'test',
+     habits: [],
+     dailys: [],
+     todos: [],
+     rewards: [],
+     leader: user._id,
+     group: "test",
+     timestamp: +(new Date),
+     members: [user],
+     official: false,
+     _isMember: true
+   });
+
+   //You can be the owner and not a member
+   var ownNotMem = new challenges.Challenge({
+     name: 'test',
+     description: 'test',
+     habits: [],
+     dailys: [],
+     todos: [],
+     rewards: [],
+     leader: user._id,
+     group: "test",
+     timestamp: +(new Date),
+     members: [],
+     official: false,
+     _isMember: false
+   });
+
+   //You not be the owner and a member
```

Comment:
I think these comments should be in this format:

> // You are the owner and a member
// You are the owner but not a member
// You are not the owner but you are a member
// You are not the owner or a member